sesman/chansrv/clipboard.c

```diff
@@ -1999,20 +1999,40 @@ clipboard_event_selection_notify(XEvent *xevent)
                 if ((g_clip_s2c.incr_in_progress == 0) && (data_size > 0))
                 {
                     g_free(g_clip_s2c.data);
+
                     g_clip_s2c.total_bytes = data_size;
                     g_clip_s2c.data = (char *) g_malloc(g_clip_s2c.total_bytes + 1, 0);
                     g_memcpy(g_clip_s2c.data, data, g_clip_s2c.total_bytes);
                     g_clip_s2c.data[g_clip_s2c.total_bytes] = 0;
                     if (g_clip_s2c.xrdp_clip_type == XRDP_CB_FILE)
                     {
-                        clipboard_send_data_response_for_file(g_clip_s2c.data,
-                                                              g_clip_s2c.total_bytes);
+                        if (g_cfg->restrict_outbound_clipboard_file)
+                        {
+                            LOG_DEVEL(LOG_LEVEL_DEBUG,
+                                      "outbound clipboard(file) UTF8_STRING(%s) is restricted because of config",
+                                      g_clip_s2c.data);
+                        }
+                        else
+                        {
+                            clipboard_send_data_response_for_file(g_clip_s2c.data,
+                                                                  g_clip_s2c.total_bytes);
+                        }
```

Comment:
I'll admit to not looking to closely at this code for now, as I've spent a while on the config stuff which I think is important to get right. We can cover this all later.

In general however, you probably want to use `LOG()` instead of `LOG_DEVEL()`. See https://github.com/neutrinolabs/xrdp/wiki/Logging for more info on the logging and please ask if you have any questions.

Reply:
https://github.com/neutrinolabs/xrdp/blob/5a11b698ef0a501d01daa2addf63b501d79935c5/sesman/chansrv/clipboard.c#L2535-L2538

It seems that the existing implementation uses `LOG_DEVEL`.
`LOG` is better for it?

Reply:
I've changed to use `LOG` in those context.

Reply:
I think LOG() is better here, but sometimes this can be a matter of opinion rather than being obvious.

As a general rule, if a system manager would find it useful use a LOG(), whereas if a developer/maintainer would find it useful, use LOG_DEVEL().

Reply:
I've created a separate PR above case only.
https://github.com/neutrinolabs/xrdp/pull/2088